### PR TITLE
Fix arm64 tick timer

### PR DESCRIPTION
**Pull request by jlaitine**
## Summary

WIP: fixed arm64 tick timer

Not going to merge before upstreaming. The first patch (arch alarm) is just cleanup to make it understandable to myself. The two others actually correct something.

### CODE REVIEW

File: arch/arm64/src/common/arm64_arch_timer.c [hunk 1]
```diff
@@ -56,6 +56,7 @@ struct arm64_oneshot_lowerhalf_s
 
   void *arg;                          /* Argument that is passed to the handler */
   uint64_t cycle_per_tick;            /* cycle per tick */
+  uint64_t next_cycle;                /* next cycle start time, i.e. compare register value */
```
Comment (jlaitine) [inline arch/arm64/src/common/arm64_arch_timer.c:59]: Actually. To simplify, I don't need to record this value. Just read the counter value every time, and align that to tick boundary.

File: drivers/timers/arch_alarm.c [hunk 2]
```diff
@@ -116,26 +116,37 @@
...
 #else
-  clock_t delta;
+  /* Start the next tick first, in order to minimize latency. Ideally
+   * the ONESHOT_TICK_START would also return the current tick so that
+   * the retriving the current tick and starting the new one could be done
+   * atomically w. respect to a HW timer
+   */
 
-  do
+  if (now - g_current_tick > 0)
```
[the remark this answers is not in the document]
Comment (jlaitine) [reply drivers/timers/arch_alarm.c:131]: Can someone please shoot the copilot? uint64_t can definitely be 0.

**Review by pussuw — APPROVED** [review]

**Comment by jlaitine** [comment]
Updated to clean cherry-picks from upstream

**Review by pussuw — APPROVED** [review]
LGTM

**Pull request merged**